src/main/java/com/mparticle/kits/ButtonKit.java

```diff
@@ -103,13 +82,53 @@ public void onResult(@Nullable Intent intent, @Nullable Throwable throwable) {
      * <li>Always include this value when reporting orders to your order API.</li>
      * </ul>
      *
+     * <pre>
+     * ButtonKit button = (ButtonKit) MParticle.getInstance()
+     *         .getKitInstance(MParticle.ServiceProviders.BUTTON);
+     * if (button != null) {
+     *     String attributionToken = button.getAttributionToken();
+     *     if (attributionToken != null) {
+     *         // Use token with your order reporting.
+     *     }
+     * }
+     * </pre>
+     *
      * @return the last tracked Button attribution token.
      **/
     @Nullable
     public String getAttributionToken() {
         return merchant.getAttributionToken(applicationContext);
     }
 
+    @Override
+    public void onAttributionTokenChanged(@NonNull String token) {
+        final Map<String, String> attributes = getIntegrationAttributes();
+        attributes.put(ATTRIBUTE_REFERRER, token);
+        setIntegrationAttributes(attributes);
+        logDebug("Refreshed Button Attribution Token: %s", token);
+    }
+
+    @Override
+    public void onResult(@Nullable Intent intent, @Nullable Throwable throwable) {
+        PackageManager pm = applicationContext.getPackageManager();
+        if (intent != null && intent.resolveActivity(pm) != null) {
+            logDebug("Handling post-install intent for %s", intent.toString());
+            AttributionResult result = new AttributionResult()
```

Comment:
This does seem to be the intended way of forwarding intent data. mParticle has a [feature branch](https://github.com/mparticle-integrations/mparticle-android-integration-button/commit/acb89cda2286763f15d85132a940d6be318a0cf0) that they committed to (but didn't merge) about a month ago. Looking at that, do you think we should add parameters that house the referrer?

[Another implementation](https://github.com/mparticle-integrations/mparticle-android-integration-branchmetrics/blob/master/src/main/java/com/mparticle/kits/BranchMetricsKit.java#L188)

Reply:
Nice find on that commit. I think this is something we can address when we submit the PR to mParticle. For now, I'm going to keep the kit communication as it was.